cluster/mesos/cluster.go

```diff
@@ -168,12 +170,20 @@ func NewCluster(scheduler *scheduler.Scheduler, TLSConfig *tls.Config, master st
 
 // Handle callbacks for the events
 func (c *Cluster) Handle(e *cluster.Event) error {
+	// publish event to the watchQueue for external subscribers
+	c.watchQueue.Publish(e)
```

Comment:
I will clean that up soon. The handler will entirely go away as it is, and only the publish set will remain.